rllib/models/torch/visionnet.py

```diff
@@ -131,7 +131,18 @@ def __init__(
                 padding, _ = same_padding(in_size, [1, 1], [1, 1])
                 if post_fcnet_hiddens:
                     layers.append(nn.Flatten())
-                    in_size = out_channels
+                    # If we have a post-fc-stack, we can allow arbitrary shapes for the final Conv2D layer.
+                    # So we use a dummy sample to determine the size of the Conv2D output (= post-fc-stack input)
+                    self._convs = nn.Sequential(*layers)
+                    # Create a B=1 dummy sample and push it through out conv-net.
+                    dummy_in = (
+                        torch.from_numpy(self.obs_space.sample())
+                        .permute(2, 0, 1)
+                        .unsqueeze(0)
+                        .float()
+                    )
+                    dummy_out = self._convs(dummy_in)
```

Comment:
sorry a bit confused too, why do you have to dry-run this conv nn to find out the output size of the last layer?
don't we know it since we just constructed layers right above?

Reply:
I must admit I'm less fluent in Torch than in TF, but my impression was that this was the most convenient way of doing it. Of course it's possible in principle to calculate it based on filter sizes, strides, and input image dimensions, but this is simpler. Would there be a reason to do a dry-run a few lines done (l187), but not here?

Reply:
ah yeah, I was just curious about it to be honest.
I'd imagine it will be safer if we specify the expected size, and the network turns out to be different. versus where we just take whatever is the output.
but I guess it's ok if this is a common practice.